tensorflow/training/docker/2.13/py3/cu118/Dockerfile.gpu

```diff
@@ -0,0 +1,386 @@
+
+# sagemaker-specific environment variable
+ENV SAGEMAKER_TRAINING_MODULE sagemaker_tensorflow_container.training:main
+
+# https://github.com/tensorflow/models/issues/9267
+# tf-models does not respect existing installations of TF and always installs open source TF
+RUN ${PIP} install --no-cache-dir -U \
+    tf-models-official==2.13.0 \
+    tensorflow-text==2.13.0
+
+RUN ${PIP} uninstall -y tensorflow tensorflow-gpu \
+  ; ${PIP} install --no-cache-dir -U \
+    ${TF_URL} \
+    tensorflow-io \
+    tensorflow-datasets
+
+RUN $PYTHON -m pip install --no-cache-dir -U \
+    numba \
+    bokeh \
+    imageio \
+    opencv-python \
+    plotly \
+    seaborn \
+    shap
+
+RUN $PYTHON -m pip install --no-cache-dir -U \
+    "sagemaker<3" \
+    sagemaker-experiments==0.* \
+    sagemaker-tensorflow-training \
+    sagemaker-training \
+    "sagemaker-studio-analytics-extension<1" \
+    "sparkmagic<1" \
+    "sagemaker-studio-sparkmagic-lib<1" \
+    smclarify
+
+# install smdebug directly the specific branch
+ARG SMDEBUG_TAG=1.0.33
+RUN git clone https://github.com/awslabs/sagemaker-debugger.git \
+    && cd sagemaker-debugger \
+    && git checkout tags/$SMDEBUG_TAG -b $SMDEBUG_TAG \
+    && $PYTHON setup.py install && cd .. && rm -rf sagemaker-debugger
+
+# install sagemaker-tensorflow from source
+# ARG SAGEMAKER_TENSORFLOW_TAG=v1.18.0
+# RUN git clone -b tf-2 https://github.com/aws/sagemaker-tensorflow-extensions.git \
+#     && cd sagemaker-tensorflow-extensions \
+#     && git checkout tags/$SAGEMAKER_TENSORFLOW_TAG -b $SAGEMAKER_TENSORFLOW_TAG \
+#     && $PYTHON -m pip install . \
+#     && cd .. && rm -rf sagemaker-tensorflow-extensions
+
+# install boost
+# tensorflow is compiled with --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=1"
+RUN wget https://sourceforge.net/projects/boost/files/boost/1.82.0/boost_1_82_0.tar.gz/download -O boost_1_82_0.tar.gz \
+   && tar -xzf boost_1_82_0.tar.gz \
+   && cd boost_1_82_0 \
+   && ./bootstrap.sh \
+   && ./b2 define=_GLIBCXX_USE_CXX11_ABI=1 threading=multi --prefix=/usr -j 64 cxxflags=-fPIC cflags=-fPIC install || true \
+   && cd .. \
+   && rm -rf boost_1_82_0.tar.gz \
+   && rm -rf boost_1_82_0 \
+   && cd /usr/include/boost
+
+# Add NGC vars
+ENV TF_AUTOTUNE_THRESHOLD=2
+
+# Remove python kernel installed by sparkmagic
+RUN /usr/local/bin/jupyter-kernelspec remove -f python3
+
+# remove tmp files
+RUN rm -rf /tmp/*
+
+RUN HOME_DIR=/root \
+   && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+   && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+   && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+   && chmod +x /usr/local/bin/testOSSCompliance \
+   && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+   && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+   && rm -rf ${HOME_DIR}/oss_compliance*
+
+CMD ["/bin/bash"]
```

Comment:
nit: Good to have an empty line at the end of a text file to prevent issues in using code for file-parsing.

Reply:
done, and fixed the 2.12 as this was a copy/update from that file.